### PR TITLE
abseil-cpp: update to 20220623

### PR DESCRIPTION
**Pull request by inglor**
Signed-off-by: Leonidas Spyropoulos <artafinde@archlinux.org>

### CODE REVIEW

**Comment by inglor** [comment]
Well not sure why VisualStudio is failing, can't replicate in linux so if anyone has an idea, please comment.

File: subprojects/packagefiles/abseil-cpp/meson.build [hunk 1]
```diff
@@ -1,5 +1,5 @@
 project('abseil-cpp', 'cpp',
-  version : '20211102.0',
+  version : '20220623.0',
   license : 'Apache-2.0',
   default_options : [
     'cpp_std=c++11',
```
Comment (neheb) [inline subprojects/packagefiles/abseil-cpp/meson.build:5]: there's your issue.

**Comment by neheb** [comment]
Fun error on clang64:

```
$ meson compile -C m1
ninja: Entering directory `C:/Users/Mangix/scoop/apps/msys2/2022-01-28/home/Mangix/devstuff/wrapdb/m1'
[6/133] Compiling C++ object subprojects/abseil-cpp-202111...libabsl_base.a.p/absl_base_internal_thread_identity.cc.obj
FAILED: subprojects/abseil-cpp-20211102.0/libabsl_base.a.p/absl_base_internal_thread_identity.cc.obj
"ccache" "c++" "-Isubprojects/abseil-cpp-20211102.0/libabsl_base.a.p" "-Isubprojects/abseil-cpp-20211102.0" "-I../subprojects/abseil-cpp-20211102.0" "-fcolor-diagnostics" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-Wnon-virtual-dtor" "-std=c++11" "-O0" "-g" "-Wno-sign-compare" "-maes" "-msse4.1" "-Wno-gcc-compat" "-pthread" -MD -MQ subprojects/abseil-cpp-20211102.0/libabsl_base.a.p/absl_base_internal_thread_identity.cc.obj -MF "subprojects/abseil-cpp-20211102.0/libabsl_base.a.p/absl_base_internal_thread_identity.cc.obj.d" -o subprojects/abseil-cpp-20211102.0/libabsl_base.a.p/absl_base_internal_thread_identity.cc.obj "-c" ../subprojects/abseil-cpp-20211102.0/absl/base/internal/thread_identity.cc
../subprojects/abseil-cpp-20211102.0/absl/base/internal/thread_identity.cc:39:1: error: unknown type name 'pthread_key_t'
pthread_key_t thread_identity_pthread_key;
^
1 error generated.
```

**Comment by neheb** [comment]
@inglor can you update this?

**Comment by inglor** [comment]
@neheb see https://github.com/mesonbuild/wrapdb/pull/932 (without extensive testing done)

**Pull request merged**